hnn_core/network.py

```diff
@@ -35,27 +37,51 @@ def read_spikes(fname, gid_dict=None):
     spike_times = []
     spike_gids = []
     spike_types = []
+    spike_tstart = []
+    spike_tstop = []
     for file in sorted(glob(fname)):
         spike_trial = np.loadtxt(file, dtype=str)
         spike_times += [list(spike_trial[:, 0].astype(float))]
         spike_gids += [list(spike_trial[:, 1].astype(int))]
 
         # Note that legacy HNN 'spk.txt' files don't contain a 3rd column for
-        # spike type. If reading a legacy version, validate that a gid_dict is
-        # provided.
-        if spike_trial.shape[1] == 3:
+        # spike type, or a 4th and 5th column for tstart and tstop. If reading
+        # a legacy version, validate that gid_dict, tstart, and tstop provided.
+        if spike_trial.shape[1] >= 3:
             spike_types += [list(spike_trial[:, 2].astype(str))]
-        else:
+        elif spike_trial.shape[1] == 2:
             if gid_dict is None:
                 raise ValueError("gid_dict must be provided if spike types "
                                  "are unspecified in the file %s" % (file,))
             spike_types += [[]]
 
-    spikes = Spikes(times=spike_times, gids=spike_gids, types=spike_types)
+        if spike_trial.shape[1] == 5:
+            spike_tstart += np.unique(spike_trial[:, 3].astype(float)).tolist()
```

Comment:
I know it's not your code but can you see if the `astype(float)` can be avoided by using the `dtype` argument in `np.loadtxt`?

Reply:
It looks like `np.loadtxt()` normally accepts a single input for `dtype`, in this case it's set to `str`. There is the option for more complex arguments but you need to know the file structure beforehand. We could potentially have two calls loading specific columns but I don't necessarily see the benefit. 

Do you mind explaining the issue with recasting to `float`'s?

Reply:
I had a similar issue when creating the write functions since there is a mixture of strings and numbers. I agree that specifying the `dtype` argument should be used from within `np.loadtxt()` whenever possible, but I'm not sure of an efficient way to use it here.